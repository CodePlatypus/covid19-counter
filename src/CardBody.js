import React, {useState} from 'react';
import PropTypes from 'prop-types';

import CountryCard from "./CountryCard";
import TextField from "@material-ui/core/TextField";

export default function CardBody(props) {
    const [filter, setFilter] = useState("");
    return (
        <div>
            <TextField id="standard-basic" label="Filter" onChange={event => setFilter(event.target.value)}/>
            {props.countriesStat.filter(countryStat => countryStat.country.toLowerCase().includes(filter.toLowerCase()))
                .map(countryStat => <CountryCard key={countryStat.country} countryStat={countryStat}/>)}
        </div>
    )
}

CardBody.prototype = {
    countriesStat: PropTypes.array.isRequired
};