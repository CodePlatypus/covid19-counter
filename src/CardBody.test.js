import React from 'react';
import {shallow} from 'enzyme';
import CardBody from './CardBody';
import toJson from 'enzyme-to-json';

test('renders CardBody with empty countriesStat', () => {
    const wrapper = shallow(<CardBody countriesStat={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

test('renders CardBody with 1 countriesStat', () => {
    const wrapper = shallow(<CardBody countriesStat={mockDataSingle} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

test('renders CardBody with 2 countriesStat', () => {
    const wrapper = shallow(<CardBody countriesStat={mockDataDouble} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

const mockDataSingle = [{
    "country": "China",
    "cases": 80967,
    "todayCases": 39,
    "deaths": 3248,
    "todayDeaths": 3,
    "recovered": 71150,
    "activeCases": 6569,
    "critical": 2136
}];

const mockDataDouble = [{
    "country": "China",
    "cases": 80967,
    "todayCases": 39,
    "deaths": 3248,
    "todayDeaths": 3,
    "recovered": 71150,
    "activeCases": 6569,
    "critical": 2136
}, {
    "country": "Italy",
    "cases": 12313,
    "todayCases": 234,
    "deaths": 123,
    "todayDeaths": 23,
    "recovered": 32523,
    "activeCases": 4324,
    "critical": 2345
}];