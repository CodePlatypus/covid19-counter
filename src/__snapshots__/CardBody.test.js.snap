// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardBody with 1 countriesStat 1`] = `
<div>
  <WithStyles(ForwardRef(TextField))
    id="standard-basic"
    label="Filter"
    onChange={[Function]}
  />
  <CountryCard
    countryStat={
      Object {
        "activeCases": 6569,
        "cases": 80967,
        "country": "China",
        "critical": 2136,
        "deaths": 3248,
        "recovered": 71150,
        "todayCases": 39,
        "todayDeaths": 3,
      }
    }
    key="China"
  />
</div>
`;

exports[`renders CardBody with 2 countriesStat 1`] = `
<div>
  <WithStyles(ForwardRef(TextField))
    id="standard-basic"
    label="Filter"
    onChange={[Function]}
  />
  <CountryCard
    countryStat={
      Object {
        "activeCases": 6569,
        "cases": 80967,
        "country": "China",
        "critical": 2136,
        "deaths": 3248,
        "recovered": 71150,
        "todayCases": 39,
        "todayDeaths": 3,
      }
    }
    key="China"
  />
  <CountryCard
    countryStat={
      Object {
        "activeCases": 4324,
        "cases": 12313,
        "country": "Italy",
        "critical": 2345,
        "deaths": 123,
        "recovered": 32523,
        "todayCases": 234,
        "todayDeaths": 23,
      }
    }
    key="Italy"
  />
</div>
`;

exports[`renders CardBody with empty countriesStat 1`] = `
<div>
  <WithStyles(ForwardRef(TextField))
    id="standard-basic"
    label="Filter"
    onChange={[Function]}
  />
</div>
`;
